voyager-lifecycle-kmp/build.gradle.kts

```diff
@@ -0,0 +1,37 @@
+plugins {
+    kotlin("multiplatform")
+    id("com.android.library")
+    id("org.jetbrains.compose")
+    id("com.vanniktech.maven.publish")
+}
+
+setupModuleForComposeMultiplatform(fullyMultiplatform = true)
+
+android {
+    namespace = "cafe.adriel.voyager.kodein"
+}
+
+kotlin {
+    sourceSets {
+        commonMain.dependencies {
+            api(projects.voyagerCore)
+            api(projects.voyagerNavigator)
+            compileOnly(compose.runtime)
+            compileOnly(compose.runtimeSaveable)
```

Comment:
I think this might lead to a warning like

```
A compileOnly dependency is used in the Kotlin/Native target 'ios....':
```

Reply:
This is a old approach that I don't know if is still required. In the past, if you not add `compileOnly` to Compose KMP libraries, then, android only project would fail to use the library because it would not find the jetbrains android dependencies. I think this is fixed, but I not 100% sure without testing, for now, I will still keep this approach.

For the warning, we have a flag
https://github.com/adrielcafe/voyager/blob/main/gradle.properties#L12-L19